litmus-portal/frontend/src/pages/EditSchedule/index.tsx

```diff
@@ -0,0 +1,504 @@
+        weightages: weightData,
+        project_id: projectID,
+        cluster_id: clusterid,
+      };
+
+      createChaosWorkFlow({
+        variables: { ChaosWorkFlowInput: chaosWorkFlowInputs },
+      });
+    }
+  };
+
+  useEffect(() => {
+    localforage.getItem('editSchedule').then((isCronEdited) => {
+      if (wfDetails !== undefined) {
+        for (let i = 0; i < wfDetails?.weightages.length; i++) {
+          w.push({
+            experimentName: wfDetails?.weightages[i].experiment_name,
+            weight: wfDetails?.weightages[i].weightage,
+          });
+        }
+        doc.contents = JSON.parse(wfDetails?.workflow_manifest);
+        workflowAction.setWorkflowManifest({
+          manifest: isCronEdited === null ? doc.toString() : manifest,
+        });
+        setWorkflow({
+          name: wfDetails?.workflow_name,
+          description: wfDetails?.workflow_description,
+        });
+        localforage.setItem('weights', w);
+        workflowAction.setWorkflowDetails({
+          workflow_id: wfDetails?.workflow_id,
+          clusterid: wfDetails?.cluster_id,
+          cronSyntax:
+            isCronEdited === null ? wfDetails?.cronSyntax : cronSyntax,
+          scheduleType: {
+            scheduleOnce:
+              wfDetails?.cronSyntax === '' ? 'now' : 'recurringSchedule',
+            recurringSchedule: '',
+          },
+          scheduleInput: {
+            hour_interval: 0,
+            day: 1,
+            weekday: 'Monday',
+            time: new Date(),
+            date: new Date(),
+          },
+        });
+      }
+    });
+
+    template.selectTemplate({ selectTemplateID: 0, isDisable: false });
+    setWeights(w);
+  }, [data]);
+
+  const tabs = useActions(TabActions);
+
+  const [yamlStatus, setYamlStatus] = React.useState(
+    'Your code is fine. You can move on!'
+  );
+  const [modified, setModified] = React.useState(false);
+
+  const handleEditOpen = () => {
+    setModified(false);
+    setOpen(true);
+  };
+
+  const handleEditClose = () => {
+    setModified(true);
+    setOpen(false);
+  };
+
+  const handleNext = () => {
+    handleMutation();
+  };
+
+  useEffect(() => {
+    const editorValidations = parseYamlValidations(manifest, classes);
+    const stateObject = {
+      markers: editorValidations.markers,
+      annotations: editorValidations.annotations,
+    };
+    if (stateObject.annotations.length > 0) {
+      setYamlStatus(`${t('createWorkflow.verifyCommit.errYaml')}`);
+    } else {
+      setYamlStatus(`${t('createWorkflow.verifyCommit.codeIsFine')}`);
+    }
+  }, [modified]);
+
+  const handleFinishModal = () => {
+    history.push({
+      pathname: `/workflows`,
+      search: `?projectID=${projectID}&projectRole=${userRole}`,
+    });
+    setFinishModalOpen(false);
+  };
+
+  const handleErrorModalClose = () => {
+    setErrorModal(false);
+  };
+
+  return (
+    <Scaffold>
+      {loading ? (
+        <Loader />
+      ) : (
+        <>
```

Comment:
is the <> needed? can we use <div> instead if yes?

Reply:
Yes a react element is required to wrap a ternary. 

`condition ? <div /> : <Button/><div />etc` is not possible unless you wrap the second part in once single React element, thus `<React Element><Button/><div />etc</React Element>` is required. Its better to use a Fragment than a div this the <></>

Reply:
can we change the logic to loading && then ? @S-ayanide

Reply:
@S-ayanide its not yet fixed.

<img width="1011" alt="Screenshot 2021-04-08 at 11 54 09 AM" src="https://user-images.githubusercontent.com/50652677/113978654-cfc55d80-9861-11eb-96d8-234d78871079.png">